src/docs/arch-storage-extension.dox

```diff
@@ -0,0 +1,107 @@
+class also provides the means to delete objects, to clean up artifacts from internal unit testing.
+
+@section extension_cloud_file_system Cloud File System Interface
+
+WiredTiger supports a database over a custom file system by exposing a \c WT_FILE_SYSTEM interface
+for the end-user to implement. The store itself provides a means to configure as many file systems
+as needed, each linked to a bucket. These file systems - \c GCP_FILE_SYSTEM, \c AZURE_FILE_SYSTEM
+and \c S3_FILE_SYSTEM - are custom implementations provided by the storage source. These filesystems
+contain an instance of the connection class, hence providing an active connection to the bucket
+embedded in the file system instance.
+
+The cloud file system uses the functionality provided by the connection class to simulate a
+write-once-read-many file system. The filesystem can list directory contents, check for a file's
+presence, and access the file by creating a file handle. Before a file handle can open a file,
+files that are not in the cache directory are downloaded from the bucket and put in the cache.
+
+The storage source uses the filesystem's put-object functionality to flush files into the cloud.
+The Tier Manager inside WiredTiger directs this flush. Only S3 storage source files pushed to the
+cloud are also copied to the local cache.
+
+@section extension_cloud_file_handle Cloud File Handle Interface
+
+Each storage source also implements the \c WT_FILE_HANDLE interface, to access the files
+on the file system. Since the object store is read-only, all the write access methods are disabled.
+
+@section extension_log_stat Messaging and Statistics
+The cloud provider SDKs use the term logging differently to WiredTiger. The SDK's logging definition
+refers to the messages that are generated. The SDKs provide their own log levels which
+are translated into WiredTiger messaging levels.
+
+All of the cloud storage sources have their own messaging system and statistics to monitor the
+state of the SDKs and the storage sources. The messaging system captures error and informational from both
+the SDK and the custom storage source which are redirected it into the WiredTiger messaging system.
+The statistics are printed when the WiredTiger verbosity is set to \c WT_VERBOSE level or higher
+on the storage source.
+
+@section s3_store S3 Storage source
+The S3 store implement the following functionalities defined by the storage source interface:
+
+- The bucket is expected to be a string argument comprised of the bucket name and the
+corresponding region separated by a ';', e.g. "bucket1;ap-southeast-2". AWS requires an access token
+to authenticate programmatic access to S3. The access token comprises an access key ID and a
+secret access key as a set. The S3 store requires the token to be provided as a single string
+argument of the access key followed by the secret key, separated by a ';'.
```

Comment:
This paragraph isn't really describing any functionality, just what the S3 store expects. If the S3 store provides thus authentication functionality, perhaps it'd be better to re-frame this section as "The S3 store handles authentication with AWS, requiring the user to provide an access token. The access token... " and then the rest of the paragraph.

Note that the initial sentence about the bucket name is gone -- that seems like a separate point, so if it's important, maybe put it in another list element.